doc/threading_model.md

```diff
@@ -0,0 +1,31 @@
+﻿# gRPC Threading Model
+
+## Introduction
+This document serves as a basic intro for the gRPC threading model. gRPC generates service code in two fashions - synchronous and asynchronous API. These two closely correlate to the threading model in play. 
+
+## Synchronous API Threading model
+This is the easiest model to start with and could be a satisfactory model for a large number of services. As such, the implementation is efficient (and is not a ‘toy’ implementation). In this model, gRPC takes care of the threading completely and the application code need to worry only about handling of the actual rpc call.
+* When the server starts up, a thread pool is created to handle incoming rpc requests. The default implementation of this thread pool is called DynamicThreadPool. It creates threads equal to number of processors in the system and all of them are in waiting state.
+* DynamicThreadPool has in­built scaling mechanism to create additional threads on demand. For integrators interested in having a little more control on the behavior of the thread pool, they can define GRPC_CUSTOM_DEFAULT_THREAD_POOL and provide the implementation of the thread pool interface (which is literally just 1 function that adds work to the pool).
+* As soon as enough data is available to process a complete rpc, the worker thread moves ahead with handling the rpc.
+* **Pros**
+	* Simple to use and get going.
+	* Efficiency would be fine for most applications. Some might gain more efficiency with a more suitable implementation of the thread pool.
+* **Cons**
+	* The application code could be called concurrently from multiple threads - for the same client or different clients.

```

Comment:
Just to be specific "The application code" -> "The application code in the service implementation"